api/types/constants.go

```diff
@@ -570,6 +570,30 @@ const (
 	// downgraded before being returned to clients on older versions that do not
 	// support one or more features enabled in that resource.
 	TeleportDowngradedLabel = TeleportInternalLabelPrefix + "downgraded"
+
+	// TeleportSystemResourceLabel marks a resource as being used internally by
+	// teleport itself, and should not be exposed to the cluster users or admins.
+	TeleportResourceTypeLabel = TeleportInternalLabelPrefix + "resource-type"
+
+	// TeleportResourceRevision marks a teleport-managed resource with a reversion
+	// number to aid future migrations. Label value is expected to be a number.
+	TeleportResourceRevision = TeleportInternalLabelPrefix + "revision"
+)
+
+// TeleportResourceType defines the expected values of the `TeleportResourceTypeLabel`
+// label.
+type TeleportResourceType string
+
+const (
+	// TeleportResourceTypeSystem marks a resource as being required internally
+	// by teleport. Teleport may reset, edit or ortherwise modify this resource
+	// at will, and no attempt to preserve any user changes will be made.
+	TeleportResourceTypeSystem TeleportResourceType = "system"
+
+	// TeleportResourceTypePreset marks a resource that is created and managed by
+	// teleport, but is open for user modification. Teleport will attempt to preserve
+	// any user modifications when changing or updating the resource.
+	TeleportResourceTypePreset TeleportResourceType = "preset"
```

Comment:
Can we backfill this label to existing resources? `access, editor, auditor`
This can be done in a follow up PR 👍